### PR TITLE
past_date validation added to the patient DOB

### LINKED ISSUE #2976: No validation on Patient DOB
### Describe the bug

A patient can have any desired DOB which should not be a case as a patient shouldn't have age in negative.
Screenshot from 2020-02-24 04-54-12

### Expected behavior
Patient DOB should always be smaller than the current date.

### Client configuration

- Browser: Chrome Version 80.0.3987.106 (Official Build) (64-bit)
- Operating system: Windows 10

### PR DESCRIPTION
**Pull request by im-Amitto**
fixes #2976 

**Previous behavior**: Patient was allowed to have a negative age

**Current behavior**: PAtient can't select the future date as DOB.

### CODE REVIEW

**Comment by stephenwaite** [comment]
first use of `past_date` for validation, nice!

File: sql/database.sql [hunk 1]
```diff
@@ -2805,7 +2805,7 @@ INSERT INTO `layout_options` (`form_id`,`field_id`,`group_id`,`title`,`seq`,`dat
...
 INSERT INTO `layout_options` (`form_id`,`field_id`,`group_id`,`title`,`seq`,`data_type`,`uor`,`fld_length`,`max_length`,`list_id`,`titlecols`,`datacols`,`default_value`,`edit_options`,`description`,`fld_rows`) VALUES ('DEM', 'pubpid', '1', 'External ID', 5, 2, 1, 10, 15, '', 1, 1, '', 'ND', 'External identifier', 0);
-INSERT INTO `layout_options` (`form_id`,`field_id`,`group_id`,`title`,`seq`,`data_type`,`uor`,`fld_length`,`max_length`,`list_id`,`titlecols`,`datacols`,`default_value`,`edit_options`,`description`,`fld_rows`) VALUES ('DEM', 'DOB', '1', 'DOB', 6, 4, 2, 10, 10, '', 1, 1, '', 'D', 'Date of Birth', 0);
+INSERT INTO `layout_options` (`form_id`,`field_id`,`group_id`,`title`,`seq`,`data_type`,`uor`,`fld_length`,`max_length`,`list_id`,`titlecols`,`datacols`,`default_value`,`edit_options`,`description`,`fld_rows`) VALUES ('DEM', 'DOB', '1', 'DOB', 6, 4, 2, 10, 10, '', 1, 1, '', 'D', 'Date of Birth', 'past_date');
```
Comment (bradymiller) [inline sql/database.sql]: the fld_rows doesn't look right since that is an int

Comment (stephenwaite) [reply sql/database.sql]: good pt, looks like all of these `layout_options` could have more fields added
```
  `list_backup_id` varchar(100) NOT NULL default '',
  `source` char(1) NOT NULL default 'F' COMMENT 'F=Form, D=Demographics, H=History, E=Encounter',
  `conditions` text COMMENT 'serialized array of skip conditions',
  `validation` varchar(100) default NULL,

Comment (bradymiller) [reply sql/database.sql]: it works if placed in the validation column :)

Comment (im-Amitto) [reply sql/database.sql]: @bradymiller 
To generate constraint for a form currently `list_options.notes` along with a few fields from `layout_options` is being used and they follow the following query for join `layout_options.validation = list_options.option_id` i.e. that's why I updated the validation field. am I missing something?

Comment (bradymiller) [reply sql/database.sql]: hi @im-Amitto , My read of that line is that you are updating the `fld_rows` field though.

Comment (bradymiller) [reply sql/database.sql]: to clarify, I agree that the validation field should be updated.

Comment (im-Amitto) [reply sql/database.sql]: I'm really sorry, I just realized it now  :expressionless:.

Comment (bradymiller) [reply sql/database.sql]: no prob at all. it is still going to win the `first use of past_date for validation` award  :)

2DV

Comment (im-Amitto) [reply sql/database.sql]: @bradymiller corrected the query :)

**Comment by bradymiller** [comment]
hi @im-Amitto , Looks good and bringing into codebase. Thanks for the fix! -brady

**Pull request merged**